test/application/use.js

```diff
@@ -52,4 +53,11 @@ describe('app.use(fn)', function(){
     .expect(404)
     .end(done);
   });
+
+  it('should throw error for non function', function(done){
+    const app = new Koa();
+
+    should(function(){ app.use('not a function'); }).throw('middleware must be a function!');
```

Comment:
+1 for that, I didn't done that in my PR because I wasn't sure if we should do it at all.
Let's open an Issue to discuss that?